packages/fuels-accounts/src/predicate.rs

```diff
@@ -40,24 +40,32 @@ impl Predicate {
     }
 
     pub fn set_provider(&mut self, provider: Provider) -> &mut Self {
+        self.address = Self::calculate_address(&self.code, &provider.consensus_parameters());
         self.provider = Some(provider);
         self
     }
 
+    pub fn calculate_address(code: &[u8], params: &ConsensusParameters) -> Bech32Address {
+        fuel_tx::Input::predicate_owner(code, params).into()
+    }
+
+    fn consensus_parameters(&self) -> ConsensusParameters {
+        self.provider()
+            .map(|p| p.consensus_parameters())
+            .unwrap_or_default()
+    }
+
+    /// Uses default `ConsensusParameters`
     pub fn from_code(code: Vec<u8>) -> Self {
         Self {
-            address: Self::calculate_address(&code),
+            address: Self::calculate_address(&code, &ConsensusParameters::default()),
```

Comment:
I think this makes way for nasty pitfalls. This means the address likely to be invalid when created this way and only becomes valid after setting a provider. If the user takes the address in the meanwhile to pass to a transfer, the funds will get lost.

Reply:
Can you please make an issue for this ?